### PR TITLE
API: return computed stats for members routes

### LINKED ISSUE #7863: /members/{uid} not returning calculated stats like maxHealth / maxMP / toNextLevel
this would be useful for API Users

as @paglias @crookedneighbor  pointed out this could be done using virtual properties

### PR DESCRIPTION
**Pull request by paglias**
Fixes #7863 

Tests are missing

### CODE REVIEW

**Comment by paglias** [comment]
Tests added, this is ready to be merged

File: test/api/v3/integration/groups/GET-groups_groupId_members.test.js [hunk 1]
```diff
@@ -57,6 +66,30 @@ describe('GET /groups/:groupId/members', () => {
     expect(res[0].profile).to.have.all.keys(['name']);
   });
 
+  it('populates only some fields if req.query.includeAllPublicFields === true and it is a guild', async () => {
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/GET-groups_groupId_members.test.js]: `req.query.includeAllPublicFields === true` -> `req.query.includeAllPublicFields !== true`

Comment (paglias) [inline test/api/v3/integration/groups/GET-groups_groupId_members.test.js]: it shows that req.query.includeAllPublicFields === true doesn't work if it's not a party

Comment (paglias) [inline test/api/v3/integration/groups/GET-groups_groupId_members.test.js]: I'll change it because it's not clear

Comment (crookedneighbor) [inline test/api/v3/integration/groups/GET-groups_groupId_members.test.js]: If that's the case, you'll need to add the query in your test case:

``` js
let res = await user.get(`/groups/${group._id}/members`);
```

Does not include the query parameter.

Comment (paglias) [inline test/api/v3/integration/groups/GET-groups_groupId_members.test.js]: Ops... Forgot about it!

**Comment by crookedneighbor** [comment]
👍

**Comment by SabreCat** [comment]
Tests aren't passing yet, though?

**Comment by paglias** [comment]
@SabreCat fixing it, the new tests are causing another one to fail

**Pull request merged**